compiler/modules/front.syntax/ast.bal

```diff
@@ -505,7 +505,7 @@ public type TypeDescRef record {|
 
 public type SingletonTypeDesc record {|
     *PositionFields;
-    (string|float|int|boolean|decimal) value;
+    SimpleConstExpr valueExpr;
```

Comment:
This is not quite right because it allows VarRefExpr.  So we need to do

```
public type ExtendedLiteralExpr LiteralExpr|NumericLiteralExpr|SimpleConstNegateExpr;
public type SimpleConstExpr ExtendedLiteralExpr|VarRefExpr;
```

and then use ExtendedLiteralExpr here.